lib/repository.py

```diff
@@ -63,7 +64,14 @@ def get_git_repository(remote_repo_url, parent_dir_path):
                       .format(name=name, url=remote_repo_url))
             repo.create_remote(MAIN_REMOTE_NAME, remote_repo_url)
         return repo
-    else:
+    except git.exc.InvalidGitRepositoryError:
+        # This exception will be thrown if the git repository has an
+        # invalid format, which can be due a previous svn repository
+        # at the repo_path directory.  In this case, we remove the old
+        # gcc repository and clone from scratch.
+        LOG.info("Repository path '%{repo_path}' has invalid format, removing "
+                 "and cloning from scratch".format(repo_path=repo_path))
+        shutil.rmtree(repo_path)
```

Comment:
A `git.exc.NoSuchPathError` could also be raised if the path does not exist. We would have to execute the same clone action in this case.

To avoid code duplication, I suggest the following structure:
```python
repo = None
try:
  repo = GitRepository(repo_path)
except git.exc.InvalidGitRepositoryError:
  remove path
except git.exc.NoSuchPathError:
  pass

if repo is None:
  return GitRepository.clone_from
else:
  create main remote
  return repo
```